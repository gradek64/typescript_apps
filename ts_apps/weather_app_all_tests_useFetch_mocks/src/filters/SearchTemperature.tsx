import React, { useState, useRef, useEffect } from 'react'

//custom 
import './searchTemperature.css'


interface IProps {
  filterCallback: (arg: [string, number]) => void
}
const SearchTemperature = ({ filterCallback }: IProps) => {

  const inputSearchRef = useRef<HTMLInputElement | null>(null);
  const [minOrMax, setMinOrMax] = useState('min')

  const onChangeMinMax = (e: React.ChangeEvent<HTMLInputElement>) => {
    const { value } = e.target as HTMLInputElement
    setMinOrMax(value)
  }

  const submitChange = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
    e.preventDefault()
    const valueToNumber = Number(inputSearchRef.current!.value)
    //simple validation for non defined
    if (!inputSearchRef!.current!.value || isNaN(valueToNumber)) return
    filterCallback([minOrMax, valueToNumber])
  }

  return (
    <form className='searchTemperature' >
      <label htmlFor="search"></label>search Temperature
      <input name='search' type="number" ref={inputSearchRef}></input>
      <div className='minMaxContainer'>
        <label htmlFor="max">min
        <input name='max' type="radio" checked={minOrMax === 'min'} onChange={onChangeMinMax} value="min" />
        </label>
        <label htmlFor="min">max
        <input name='min' type="radio" checked={minOrMax === 'max'} onChange={onChangeMinMax} value="max" />
        </label>
      </div>
      <button id='submit' onClick={submitChange} >Search</button>
    </form>
  )
}


export default SearchTemperature