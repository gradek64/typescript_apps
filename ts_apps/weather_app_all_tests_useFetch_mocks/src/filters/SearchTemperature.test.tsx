import { fireEvent, render } from '@testing-library/react';
import SearchTemperature from './SearchTemperature';


describe('@render', () => {
  it('Component has loaded data', async () => {
    const { asFragment } = render(<SearchTemperature filterCallback={() => { }} />);
    expect(asFragment()).toMatchSnapshot();
  });
})


