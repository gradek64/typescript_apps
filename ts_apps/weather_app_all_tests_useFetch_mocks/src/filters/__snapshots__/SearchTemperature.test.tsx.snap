// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@render Component has loaded data 1`] = `
<DocumentFragment>
  <form
    class="searchTemperature"
  >
    <label
      for="search"
    />
    search Temperature
    <input
      name="search"
      type="number"
    />
    <div
      class="minMaxContainer"
    >
      <label
        for="max"
      >
        min
        <input
          checked=""
          name="max"
          type="radio"
          value="min"
        />
      </label>
      <label
        for="min"
      >
        max
        <input
          name="min"
          type="radio"
          value="max"
        />
      </label>
    </div>
    <button
      id="submit"
    >
      Search
    </button>
  </form>
</DocumentFragment>
`;
